app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -77,16 +78,17 @@ protected void log(int priority, String tag, String message, Throwable throwable
         Database db = helper.getWritableDb();
         daoSession = new CustomDaoMaster(db).newSession();
 
-        VersionHelper.updateAppVersion(getApplicationContext());
+        int oldVersionCode = AppPrefs.getAppVersionCode();
+        if (oldVersionCode < 2000010) {
 
-        /**
-         * Check and delete existing APKs to avoid manually deleting corrupt files before upgrading to version 2.0.9
-         * of the Appstore application
-         */
-        int newVersionCode = VersionHelper.getAppVersionCode(context);
-        if (newVersionCode < 2000009) {
+            /**
+             * Check and delete existing APKs to avoid manually deleting corrupt files before upgrading to version 2.0.9
+             * of the Appstore application
+             */
             deleteExistingApks();
         }
+
+        VersionHelper.updateAppVersion(getApplicationContext());
```

Comment:
Should this be executed if there is no upgrade happening? It seems like this line of code will trigger on every application startup?

Reply:
Correct. I mentioned the case above.